src/PlaygroundGrains/ProcessGrain.cs

```diff
@@ -0,0 +1,60 @@
+using Orleans.Concurrency;
+using GrainInterfaces;
+
+// https://github.com/dotnet/orleans/issues/6389#issuecomment-597567547
+namespace Grains;
+
+[StatelessWorker]
+public class ProcessGrain<TRequest, TResponse> : Grain, IProcessGrain<TRequest, TResponse>
+{
+    protected Task<TResponse?> _myLongRunningTask;
+    protected TRequest _request;
+    protected CancellationTokenSource _cancellation = new CancellationTokenSource();
+
+    public Task StartAsync(TRequest request)
+    {
+        _request = request;
+        // option 1: let it run in the current grain activation scheduler
+        // _myLongRunningTask = Task.Factory.StartNew(_ => DoLongRunningWorkAsync(), null, _cancellation.Token, TaskCreationOptions.DenyChildAttach, TaskScheduler.Current).Unwrap();
+
+        // option 2: make it run on the thread pool scheduler
+        _myLongRunningTask = Task.Run(() => DoLongRunningWorkAsync(), _cancellation.Token);
```

Comment:
Since Grains die on its own on idle, we probably need to make sure that the grains don't idle and die before LongRunningWorkAsync ends.

Reply:
Polling to ensure grains don't die: https://github.com/dotnet/orleans/issues/6389#issuecomment-597567547